docs/requests-new.md

```diff
@@ -0,0 +1,1941 @@
+        - ED25519: ed25519 signature
+        - ED25519_MULTI: ed25519 signature in multisig case.
+    
+    - `values` (list): 
+        
+        - `from` (base58-encoded string):
+        Identifier (DID) of signer as base58-encoded string for 16 or 32 bit DID value.
+        
+        - `value` (base58-encoded string):
+         signature value
+    
+- `serialization` (string enum, optional):
+
+    Defines how the `msg` is serialized
+     - JSON: json
+     - MSG_PACK: msgpack
+        
+- `msg` (dict):
+    
+    Serialized message.
+    
+    
+## Common Request Structure
+
+Each Request (both write and read) follows the pattern as shown above.
+
+```
+{
+    "type": <...>,
+    "ver": <...>,
+    "protocolVersion": <...>,
+    
+    "data": {
+        "ver": <...>,
+        <msg-specific fields>
+    },
+    
+    "metadata": {
+        "reqId": <...>,
+        "from": <...>,
+    },
+}
+```
+
+- Message Type `type` is one of the following values:
+
+    - NODE = 0
+    - NYM = 1
+    - ATTRIB = 100
+    - SCHEMA = 101
+    - CLAIM_DEF = 102
+    - POOL_UPGRADE = 109
+    - NODE_UPGRADE = 110
+    - POOL_CONFIG = 111
+    - GET_TXN = 3
+    - GET_ATTR = 104
+    - GET_NYM = 105
+    - GET_SCHEMA = 107
+    - GET_CLAIM_DEF = 108 
+
+- `metadata` (dict):
+
+    Metadata coming with the Request and saving in the transaction as is (if this is a write request).
+
+    - `from` (base58-encoded string):
+         Identifier (DID) of the transaction submitter (client who sent the transaction) as base58-encoded string
+         for 16 or 32 bit DID value.
+         It must be present on Ledger for write requests and can be any value for read requests.
+         
+         It may differ from `did` field for some of requests (for example NYM), where `did` is a 
+         target identifier (for example, a newly created DID identifier).
+         
+         *Example*: `from` is a DID of a Trust Anchor creating a new DID, and `did` is a newly created DID.
+         
+    - `reqId` (integer): 
+        Unique ID number of the request with transaction.
+        
+- Please find the format of each request-specific data for each type of request below.
+
+## Common Reply Structure
+
+Each Write/Read Reply follows the pattern as shown above.
+
+```
+{
+    "type": REPLY,
+    "ver": <...>,
+    "protocolVersion": <...>,
+    
+    "data": {
+        "ver": <...>,
+        "results": [
+            "result": {
+                <result>
+            },
+            
+            "multiSignature": {
+                "type": <...>,
+                "value": <...>,
+                "from": <...>,
+                "serialization": <...>,
+                "signedState": <...>
+            }, 
+        
+            "stateProof": <...>,
+            "auditProof": <...>, 
+        ]
+    },
+    "metadata": {
+        "reqId": <...>,
+        "from": <...>,
+    },
+}
+```
+where `multiSignature`'s `signedState` is a serialized value having the following form:
```

Comment:
@AlexanderShekhovcov Where is the actual serialised data that was signed? Or does the sdk know how to serialise the data

Reply:
This is the data serialized and signed by nodes. So, the serialized data is signed, not actual one.
This data is not re-serialized on nodes side and passed to clients in serialized format. So, the client can verify the signature over serialized data.
Then it can de-serialize the data according to the serialization specified in "serialization" field.

BTW the same approach is going to be applied for Requests, where signature is calculated against serialized data, and the serialized data is passed to the pool, not actual one, and re-serializations on node side for requests is not needed.

Reply:
But what is the data that is serialised? `signedState` just has 2 metadata objects, eg. in a `GET_NYM` txn, the serialised data will have the actual NYM and verkey, does that not live under `signedState`?

Reply:
Ok, can you point out where is the data that is serialised for verifying the proof?

Reply:
Data used to verify BLS multisig is in `signedState `
Data used to verify state proof (I think you mean the real result for the get request including key and value in state trie) is in <result>.

Reply:
Sorry, in <result>